### PR TITLE
Do not force GcStress=0xc for GitHub_23199 on OSX.

### PR DESCRIPTION
**Pull request by sandreenko**
There was a typo for Unix (it is COMPlus_GCStress, not COMPlus_GcStress, but COMPlus_GcStressOnDirectCalls was correct). That was why the test passed on OSX.

### CODE REVIEW

File: tests/src/JIT/Regression/JitBlue/GitHub_23199/GitHub_23199.csproj [hunk 1]
```diff
@@ -29,16 +29,17 @@
   <ItemGroup>
     <Service Include="{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}" />
   </ItemGroup>
-  <PropertyGroup>
+  <!-- OSX doesn't support GCStress=0xc. -->
+  <PropertyGroup Condition="'$(__BuildOS)' != 'OSX'">
```
Comment (jashook) [inline tests/src/JIT/Regression/JitBlue/GitHub_23199/GitHub_23199.csproj]: I think we should prefer <DisableBuild Condition="'$(__BuildOS)' != 'OSX'" />

**Comment by sandreenko** [comment]
@jashook I have changed my mind and decided not to set GCStress at all. You were right, it is too dangerous to run them in outerloop, I saw unstable GCStress infra failures that we do not want to see in outerloop jobs. Also, gcstress is not supported on OSX and Alpine and it makes these conditions worse.

So it is better not to force it at all as you initially suggested. We will still have this coverage in GCStress pipeline. Hope to delete the setting of `COMPlus_GcStressOnDirectCalls` in the future.

**Comment by jashook** [comment]
@sandreenko I agree with the decision to remove gcstress from outerloop. Thanks for the change!

**Comment by mjsabby** [comment]
Your PR was completed with a merge commit. The repo guideline is to use squash and merge.

**Comment by sandreenko** [comment]
> Your PR was completed with a merge commit. The repo guideline is to use squash and merge.

Thanks, forgot that. 
However I do not see this guideline in the repro guideline https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/contributing.md. Do you want to make a PR and add it?

**Comment by AndyAyersMS** [comment]
Are there issues somewhere describing why GC stress is not supported on OSX or Alpine?

**Comment by jashook** [comment]
OSX: https://github.com/dotnet/coreclr/issues/5761

**Comment by sandreenko** [comment]
> Are there issues somewhere describing why GC stress is not supported on OSX or Alpine?

On Alpine we do not have coredistools, we did not build them there and we can't use package that was built on normal Linux.

**Pull request merged**